pkg/virtctl/vnc/vnc.go

```diff
@@ -155,70 +155,71 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 		fd, err := ln.Accept()
 		if err != nil {
 			log.Log.V(2).Infof("Failed to accept unix sock connection. %s", err.Error())
-			listenResChan <- err
+			errChan <- err
+			return
 		}
 		defer fd.Close()
 
-		log.Log.V(2).Infof("VNC Client connected in %v", time.Now().Sub(start))
+		log.Log.V(2).Infof("VNC Client connected in %v", time.Since(start))
 		templates.PrintWarningForPausedVMI(virtCli, vmi, namespace)
 
 		// write to FD <- pipeOutReader
 		go func() {
 			_, err := io.Copy(fd, pipeOutReader)
-			readStop <- err
+			errChan <- err
 		}()
 
 		// read from FD -> pipeInWriter
 		go func() {
 			_, err := io.Copy(pipeInWriter, fd)
-			writeStop <- err
+			errChan <- err
 		}()
 
 		// don't terminate until vnc client is done
-		<-doneChan
-		listenResChan <- err
+		<-ctx.Done()
 	}()
 
 	port := ln.Addr().(*net.TCPAddr).Port
 
 	if proxyOnly {
-		defer close(doneChan)
 		optionString, err := json.Marshal(struct {
 			Port int `json:"port"`
 		}{port})
 		if err != nil {
-			return fmt.Errorf("Error encountered: %s", err.Error())
+			return fmt.Errorf("error encountered: %s", err.Error())
 		}
 		fmt.Fprintln(cmd.OutOrStdout(), string(optionString))
 	} else {
 		// execute VNC Viewer
-		go checkAndRunVNCViewer(doneChan, viewResChan, port)
+		go checkAndRunVNCViewer(ctx, errChan, port)
 	}
 
 	go func() {
-		defer close(stopChan)
 		interrupt := make(chan os.Signal, 1)
 		signal.Notify(interrupt, os.Interrupt)
-		<-interrupt
+		defer signal.Stop(interrupt)
+
+		select {
+		case <-interrupt:
```

Comment:
Another one last question (hopefully): Do we even need a separate goroutine for this? Can we put `case <-interrupt` into our select below (L210)?